src/middleware/cloudWatch.ts

```diff
@@ -37,41 +34,61 @@ export class CloudWatchMiddleware
       .body as WrappedDocument<HealthCertDocument>;
     const data = getData(wrappedDocument);
     const provider = data.issuers[0].identityProof?.location ?? "UNKNOWN";
-    this.provider = provider;
-    const subDomain = this.extractSubDomain(provider);
-    trace(`provider ${provider} attempting to notarise pdt...`);
-    trace(`subDomain ${subDomain} attempting to notarise pdt...`);
+    this.specificDomain = provider;
+    const aggregateDomain = this.toAggregateDomain(this.specificDomain);
+    trace(
+      `specificDomain ${this.specificDomain} attempting to notarise pdt...`
+    );
+    trace(`aggregateDomain ${aggregateDomain} attempting to notarise pdt...`);
   };
 
   after = async (req: Request): Promise<void> => {
-    const { body } = req.response;
+    const { body, statusCode } = req.response;
+    if (statusCode !== 200) {
+      logError(
+        "error encountered, logging for successful notarisation skipped"
+      );
+      return;
+    }
+
     try {
       const notarisationResult: NotarisationResult = JSON.parse(body);
       const { notarisedDocument } = notarisationResult;
       const data: HealthCertDocument | PDTHealthCertV2Document =
         getData(notarisedDocument);
-      let testName = "";
-      if (data?.version === "pdt-healthcert-v2.0") {
-        // @ts-ignore
-        const observationResource = data.fhirBundle.entry?.find(
-          (entr: any) => entr.resource.resourceType === "Observation"
-        )?.resource;
-        const observation = parsers(observationResource) as ObservationV2;
-        testName = observation.testType?.display || "";
+      let testType = "";
+      const validTestTypes = ["art", "pcr", "ser"];
+      if (data.version === "pdt-healthcert-v2.0") {
+        // version 2
+        testType = (data.type as string).toLowerCase();
```

Comment:
in v2, this type will be string or array. So, now it's throw error when i tried with `['PCR', 'SER']` type. you can try with `test/fixtures/v2/pdt_pcr_ser_with_multi_result_wrapped.json`

Reply:
ok good catch